test/AsmResolver.PE.Tests/DotNet/Metadata/MetadataTest.cs

```diff
@@ -198,5 +203,51 @@ public void SelectFirstUserStringsStreamInEnCMetadata()
         {
             AssertCorrectStreamIsSelected<UserStringsStream>(Properties.Resources.HelloWorld_DoubleUserStringsStream_EnC, true);
         }
+
+        [Fact]
+        public void SchemaStreamShouldForceEnCMetadata()
+        {
+            var peImage = PEImage.FromBytes(Properties.Resources.HelloWorld_SchemaStream);
+            AssertCorrectStreamIsSelected<BlobStream>(peImage, true);
+            AssertCorrectStreamIsSelected<GuidStream>(peImage, true);
+            AssertCorrectStreamIsSelected<StringsStream>(peImage, true);
+            AssertCorrectStreamIsSelected<UserStringsStream>(peImage, true);
+        }
+
+        [Fact]
+        public void UseCaseInsensitiveComparisonForHeapNamesInEnCMetadata()
+        {
+            var peImage = PEImage.FromBytes(Properties.Resources.HelloWorld_LowerCaseHeapsWithEnC);
+            var metadata = peImage.DotNetDirectory!.Metadata!;
+
+            Assert.True(metadata.TryGetStream(out BlobStream? _));
+            Assert.True(metadata.TryGetStream(out GuidStream? _));
+            Assert.True(metadata.TryGetStream(out StringsStream? _));
+            Assert.True(metadata.TryGetStream(out UserStringsStream? _));
+        }
+
+        [Fact]
+        public void UseLargeTableIndicesWhenJTDStreamIsPresentInEnCMetadata()
```

Comment:
This also needs a writer test that validates the right column sizes are used (e.g., reread rebuilt binary and test column sizes).